src/nn/Modules/Activations.cpp

```diff
@@ -46,5 +46,48 @@ namespace af
         {
             return max(input, m_slope * input);
         }
+
+        PReLU::PReLU(int size, double spread)
+        {
+            auto w = nn::weight(size, 1, spread);
+            setParams({w});
+        }
+
+        PReLU::PReLU(const Variable &w) :
+            Module({w})
+        {
+        }
+
+        Variable PReLU::forward(const Variable &input)
+        {
+            auto tmp = max(input, 0.0);
+            auto res = expandAs(m_parameters[0],tmp) * tmp;
+            //TODO: Determine if doing the max after the mul is preferable
```

Comment:
That would be two different behaviors. max after multiply will have a different behavior when values of `input` and `m_parameters` are both negative. Can you double check what the exact behavior should be.